File: LeetCode/169. Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {

  unordered_map<int, int> map;
  for ( int i = 0; i < nums.size(); i++ ) 
    map[nums[i]] ++;

  int n = 0;
  int max = 0;
  for ( auto i : map ) {
    if ( max < i.second ) {
      max = i.second;
      n = i.first;
    }
  }
  return n;
     
    }
};

File: LeetCode/169. Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
    HashMap<Integer, Integer> map = new HashMap<>();
    for ( int i = 0; i < nums.length; i++ ) 
      map.put ( nums[i], map.getOrDefault(nums[i], 0) + 1 );

    int n, max;
    n = max = 0;
    for ( Map.Entry<Integer, Integer> i : map.entrySet() ) 
      if ( max < i.getValue() ) {
        max = i.getValue();
        n = i.getKey();
      }
    return n;

    }
}
